x-pack/plugins/triggers_actions_ui/public/application/sections/rule_details/components/rule_event_log_list_status_filter.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState, useCallback } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { EuiFilterButton, EuiPopover, EuiFilterGroup, EuiFilterSelectItem } from '@elastic/eui';
+import { EcsEventOutcome } from 'kibana/server';
+import { RuleEventLogListStatus } from './rule_event_log_list_status';
+
+const statusFilters: EcsEventOutcome[] = ['success', 'failure', 'unknown'];
+
+interface RuleEventLogListStatusFilterProps {
+  selectedOptions: string[];
+  onChange: (selectedValues: string[]) => void;
+}
+
+export const RuleEventLogListStatusFilter = (props: RuleEventLogListStatusFilterProps) => {
+  const { selectedOptions = [], onChange = () => {} } = props;
+
+  const [isPopoverOpen, setIsPopoverOpen] = useState<boolean>(false);
+
+  const onFilterItemClick = useCallback(
+    (newOption: string) => () => {
+      if (selectedOptions.includes(newOption)) {
+        onChange(selectedOptions.filter((option) => option !== newOption));
+        return;
+      }
+      onChange([...selectedOptions, newOption]);
+    },
+    [selectedOptions, onChange]
+  );
+
+  const onClick = useCallback(() => {
```

Comment:
Yes, it's used to prevent unnecessary re-renders, normally you'd be correct in that we can just bind the `setState`, but in this case, the event handler needs the previous value so we need so I thought I'd move it into another function and memoize it

Reply:
Fair point, but I'm trying to understand the difference in UX here. I've changed the code locally to not use `useCallback` and the UI seems to respond as expected. What would you expect to see with `useCallback` removed?

Reply:
with it removed, you'd have to bind it as `onClick={() => setIsPopoverOpen(!isPopoverOpen)}` since the onClick passes the click event is its argument